addons/mail/static/src/core/common/thread.xml

```diff
@@ -23,7 +23,7 @@
                         <DateSection date="msg.dateDay" className="'pt-2'"/>
                         <t t-set="currentDay" t-value="msg.dateDay"/>
                     </t>
-                    <div t-if="isNewMessageSeparatorVisible(msg, prevMsg)" class="o-mail-Thread-newMessage d-flex align-items-center fw-bolder">
+                    <div t-if="msg.threadAsFirstUnread" class="o-mail-Thread-newMessage d-flex align-items-center fw-bolder">
```

Comment:
It feels slightly unnatural to use `threadAsFirstUnread` field here. I'd prefer `msg.eq(props.thread.firstUnreadMessage)`. Not a huge deal though

Reply:
This will trigger re-render every time `firstUnreadMessage` changes.
It's better to use the compute/inverse as it is cached.

Reply:
Oh yeah right... ok then